File: 0002-add-two-numbers/0002-add-two-numbers.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
        let carry = 0;
    let result = new ListNode(-1);
    let dummy = result;
        
    while (l1 || l2 || carry){
        let val1 = l1 ? l1.val : 0;
        let val2 = l2 ? l2.val : 0;
        let sum = (val1 + val2 + carry);
        result.next = new ListNode(sum % 10);
        result = result.next 
        carry = Math.floor(sum /10)
        l1 = l1 ? l1.next: null;
        l2 = l2 ? l2.next: null;
        console.log(result)
        console.log(dummy)
    }
    return dummy.next
};

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        carry = 0
        dummy = ListNode(0)
        curr = dummy
        while l1 or l2:
            digit = carry 
            carry = carry
            if l1:
                digit += l1.val
                carry += l1.val
            if l2:
                digit += l2.val
                carry += l2.val
            
            digit = digit % 10
            carry = carry // 10
            print(digit, carry)
            curr.val = digit
            if l1 and l1.next:
                curr.next = ListNode()
            if l2 and l2.next:
                curr.next = ListNode()
            if carry:
                curr.next = ListNode(carry)
            
            curr = curr.next 
            l1 = l1.next if l1 else None  
            l2 = l2.next if l2 else None
            
        return dummy